terraform/api.tf

```diff
@@ -0,0 +1,261 @@
+resource "aws_api_gateway_deployment" "prod" {
+  rest_api_id = aws_api_gateway_rest_api.kuberoke.id
+
+  triggers = {
+    redeployment = sha1(jsonencode(aws_api_gateway_rest_api.kuberoke.body))
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_api_gateway_stage" "prod" {
+  deployment_id = aws_api_gateway_deployment.prod.id
+  rest_api_id   = aws_api_gateway_rest_api.kuberoke.id
+  stage_name    = "kuberoke-${var.environment}-prod"
+}
+
+resource "aws_lambda_permission" "registrations" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.api_registration_handler.function_name
+  principal     = "apigateway.amazonaws.com"
+}
+
+resource "aws_api_gateway_rest_api" "kuberoke" {
+  body = jsonencode({
+    "openapi" : "3.0.1",
+    "info" : {
+      "title" : "kuberoke-${var.environment}",
+      "version" : "2023-09-26T19:48:16Z"
+    },
+    "paths" : {
+      "/reservation" : {
+        # "get" : {
```

Comment:
this is used by the admin front end to list all invited users (does a full tablescan on initial load)